compiler/packages/babel-plugin-react-compiler/src/HIR/FindContextIdentifiers.ts

```diff
@@ -126,7 +126,7 @@ function handleIdentifier(
 function handleAssignment(
   currentFn: BabelFunction | null,
   identifiers: Map<t.Identifier, IdentifierInfo>,
-  lvalPath: NodePath<t.LVal>,
+  lvalPath: NodePath<t.LVal | t.OptionalMemberExpression>,
```

Comment:
same here

Reply:
Should callers of this throw in the same way as lowerAssignment?

Reply:
yes please!

Reply:
Updated to throw at the call site